cloud_governance/policy/aws/spot_savings_analysis.py

```diff
@@ -0,0 +1,107 @@
+
+from datetime import datetime, timedelta
+
+from cloud_governance.common.clouds.aws.athena.AthenaOperations import AthenaOperations
+from cloud_governance.common.elasticsearch.elasticsearch_operations import ElasticSearchOperations
+from cloud_governance.common.logger.init_logger import logger
+from cloud_governance.common.pandas.pandas_operations import PandasOperations
+from cloud_governance.main.environment_variables import environment_variables
+
+
+class SpotSavingsAnalysis:
+    """
+    This class contain the spt savings analysis reports from the athena query
+    that are gathered from the AWS Cost and Usage Reports.
+    To get reports from the athena:
+    1. Enable the cost-and-usage reports with support of athena integration
+    2. Create a Database and table of CUR
+    """
+
+    def __init__(self):
+        self.__environment_variables_dict = environment_variables.environment_variables_dict
+        self.__default_round_digits = self.__environment_variables_dict.get('DEFAULT_ROUND_DIGITS')
+        self.__es_index = self.__environment_variables_dict.get('es_index')
+        self.__database_name = self.__environment_variables_dict.get('ATHENA_DATABASE_NAME')
+        self.__table_name = self.__environment_variables_dict.get('ATHENA_TABLE_NAME')
+        self.__athena_operations = AthenaOperations()
+        self.__es_operations = ElasticSearchOperations()
+
+    def __get_prepared_query(self):
+        """
+        This method prepare the query
+        :return:
+        """
+        current_date = datetime.utcnow()
+        year = current_date.year
+        current_month = current_date.month
+        previous_month = current_month - 1 if current_month - 1 != 0 else 12
+        query = f"""
```

Comment:
I think the query in sensitive data and we should put it under environment variable